### DISCUSSION TITLE
Extra values from getValue() after migrating to v7

### DISCUSSION THREAD

**Discussion by kashif-nazar** [react-hook-form/react-hook-form#6192] [General]
We have upgraded our application from RHF v6 to v7. One change that we are noticing is that when we call `getValues`, we get some "extra" values. 

For example, we if we have a controlled field for `name` in our form but we call  `reset({ name: "John", age: 20})` and then we call `getValues()` it returns us the value: `{ name: "John", age: 20}`. I understand that this makes more sense but since our application was build using v6, we need to our forms to only return the value that have a controlled field. e.g., `{ name: 'John' }` in this case.

What is the best way to go around this?

**Comment by kashif-nazar** [comment]
This example shows the difference of behavior. If we alert all values on v6 and v7 of this example, the output is different.
https://codesandbox.io/s/react-hook-form-v7-getvalues-forked-eskky

**Reply by bluebill1049** [reply]
in v7, getValues() will do a shallow merge with `defaultValues`, this is due to a high request for getting `defaultValues` with `getValues`.
